build-tools/automation/build.groovy

```diff
@@ -89,8 +109,20 @@ timestamps {
             sh "rm -rf \$HOME/.android/avd/XamarinAndroidTestRunner.*"
         }
 
-        utils.stageWithTimeout('prepare deps', 30, 'MINUTES', XADir, true) {    // Typically takes less than 2 minutes
+        utils.stageWithTimeout('prepare deps', 30, 'MINUTES', XADir, true) {    // Typically takes less than 2 minutes, but can take longer if any prereqs need to be provisioned
+            if (isCommercial) {
+                sh "make prepare-external-git-dependencies"
+
+                utils.stageWithTimeout('provisionator', 30, 'MINUTES', "${commercialPath}/build-tools/provisionator", true) {
+                    sh('./provisionator.sh profile.csx -v')
+                }
+            }
+
             sh "make prepare-deps CONFIGURATION=${env.BuildFlavor} V=1 MSBUILD_ARGS='$EXTRA_MSBUILD_ARGS'"
+
+            if (isCommercial) {
+                sh "make prepare-image-dependencies CONFIGURATION=${env.BuildFlavor} V=1 MSBUILD=msbuild MSBUILD_ARGS='$EXTRA_MSBUILD_ARGS'"
```

Comment:
This is something that I inherited from an old azure pipeline build but I am not sure if it's needed. @jonpryor is this something that should be included here? If so, should we also update https://github.com/xamarin/xamarin-android/blob/master/Documentation/building/unix/instructions.md#building-xamarinandroid to mention it?

Reply:
See also aa9b33a4d1729672af6dc588c17f5eff5b708179.  The purpose behind `make prepare-image-dependencies` is to generate a shell script to be run on "clean" machines to install build dependencies *before* imaging. The imagined/theoretical workflow was:

 1. Run `make prepare-image-dependencies` to generate `prepare-image-dependencies.sh`
 2. On some "clean" VM, run `prepare-image-dependencies.sh`
 3. Save the state of (2) "somewhere"
 4. Future VM builds would use (3) instead of being "really" clean.

That said... This was largely theoretical.  I don't believe any such envisioned build environment was *actually* constructed or used.

It would thus be *probably sane* to just skip this target.

*or* to *always* execute this target.

Reply:
Given this context and the fact, at least for now, that the builds run against dedicated, fully-configured hardware we don't need this step.  Removed with commit https://github.com/xamarin/xamarin-android/pull/3004/commits/0ae06c6c93d709872fb92bf2ede824a96f07ea51